### PR TITLE
chore: hardcoding open-source cloudscape favicon

### PR DESCRIPTION
**Pull request by dpitcock**
This pull request adds the cloudscape favicon from cloudscape.design site ensuring that we show an approved version into our open source package, and adds it to our webpack html plugin, where we pack will then add it to the dev server during integ tests.

By having a favicon in the index file, there is less likeliness for failed integ test from the SEVERE network error of a missing favicon, especially when chrome versions are updated.

`Unexpected errors in browser console:
    [
      {
        "level": "SEVERE",
        "message": "http://localhost:8080/favicon.ico - Failed to load resource: the server responded with a status of 404 (Not Found)",
        "source": "network",
        "timestamp": 1724677252544
      }
    ]`

This favicon will make it harder for us to distinguish our dev and even deployed environments in the browser's fabs, however future followups where different colored favicons will be added once available 

### Description

Related links, issue #, if available: n/a

### How has this been tested?

Open https://d21d5uik3ws71m.cloudfront.net/components/8e7f0dc2b7c184cb3cfc92fbe8a6e0909db20553/dev-pages/index.html#/light/ on different browsers

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by just-boris** [comment]
Any chances of making it more clean and readable? There is `favicon` option in the HtmlWebpackPlugin which we already use

File: pages/app/index.html [hunk 1]
```diff
@@ -4,6 +4,8 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <!-- This favicon gets overritten via webpack-->
```
Comment (just-boris) [inline pages/app/index.html:7]: Ran locally, there are two icons now
image

So, the icon does not get overwritten and also the new icon is after the CSP generator, so there are CSP errors in your output too

Comment (dpitcock) [reply pages/app/index.html:7]: When there is 2, this gives the browsers the ability to pick a winner.  So one webpack and one base64 as backup? I will remove the non webpack. I am not able to reproduce the CSP error locally
Screenshot 2024-08-28 at 15 08 00

Comment (just-boris) [reply pages/app/index.html:7]: I do not think you can control the injection order in the plugin, it always appends stuff to the end of the `<head>`.

So, the `favicon` prop won't work, it needs manual template:

```
<link rel="icon" type="image/png" sizes="16x16" href="data:;<%= htmlWebpackPlugin.options.faviconBase64 %>" />
```

and then pass it via

```
new HtmlWebpackPlugin({
   ...
  faviconBase64: fs.readFileSync('favion.ico', 'base64')
}),
```

**Review by abdhalees — APPROVED** [review]

File: pages/app/index.html [hunk 2]
```diff
@@ -43,3 +43,6 @@
     </div>
   </body>
 </html>
+
```
Comment (orangevolon) [inline pages/app/index.html:46]: Are these three empty lines intentional?

**Comment by dpitcock** [comment]
resolution no longer needed. We can reopen if needed at another time

**Pull request closed without merging**